ocaml/xapi/xapi_host_helpers.ml

```diff
@@ -204,6 +204,22 @@ let assert_startup_complete () =
     (fun () -> if not (!startup_complete) then
         raise (Api_errors.Server_error (Api_errors.host_still_booting, [])))
 
+let assert_host_pviommu_ready =
+  let is_ready = ref false in
+  fun () ->
+    (if not !is_ready then
+       try
+         let s = Unixext.string_of_file "/sys/kernel/pv_iommu_ready" in
+         let s = Xstringext.String.strip Xstringext.String.isspace s in
+         is_ready := int_of_string s = 1;
```

Comment:
I think this could be done simpler:
```
let is_ready =  
  try 
    let s = Unixext.string_of_file "/sys/kernel/pv_iommu_ready" in
    Scanf.sscanf " %d " (fun d -> d=1)  
  with _ -> true
```

Note that this will scan strings like "1" or " 1" successfully.

Reply:
Updated with scanf function suggested. 
Though the mutable variable `is_ready` is on purpose, to avoid re-reading/re-parsing once it already becomes `true`.